library/src/fy_library/flows/create_template_model_using_parsed_fy_py_file/main_fy.py

```diff
@@ -78,8 +76,14 @@ def __init__(
 
     def __call__(self) -> BaseTemplateModel:
         # fy:end <<<===
-        return (
-            self._optional_base_flow_template_model_with_property_setters
-            or self._optional_flow_template_model_with_property_setters
-            or self._parsed_fy_py_file.template_model
-        )
+        match self._parsed_fy_py_file.file_type:
```

Comment:
The main flow will only have a `match/case` branching to invoke particular flow.